### PR TITLE
Better ethereum enable error handling

### PR DESCRIPTION
**Pull request by tomclose**
Work towards fixing https://github.com/statechannels/monorepo/issues/1240. Probably worth merging now, as it includes work on the json-rpc error format, which will be useful elsewhere.

### CODE REVIEW

**Comment by tomclose** [comment]
New screens: 

Screen Shot 2020-03-19 at 12 27 44 PM

Screen Shot 2020-03-19 at 12 28 25 PM

**Comment by geoknee** [comment]
New screens look good: is the alignment a bit off on the first one?

**Comment by tomclose** [comment]
Yes :) - and I tried to fix it for a while, but then decided it wasn't the highest priority ...

File: packages/xstate-wallet/src/messaging.ts [hunk 1]
```diff
@@ -85,6 +87,11 @@ export class MessagingService implements MessagingServiceInterface {
     this.eventEmitter.emit('SendMessage', response);
   }
 
+  public async sendError(id: number, error: ErrorResponse['error']) {
+    const response = {id, jsonrpc: '2.0', error} as ErrorResponse; // typescript can't handle this otherwise
```
Comment (geoknee) [inline packages/xstate-wallet/src/messaging.ts]: Any idea why?

Comment (andrewgordstewart) [reply packages/xstate-wallet/src/messaging.ts]: It's probably the `'2.0'` being typed as `string` in `response`.
```suggestion
    const response: Response = {id, jsonrpc: '2.0', error};
```

If you assign the type `Response` to `response`, it has a similar effect while providing type safety -- `'1.0'` is _not_ assignable to `'2.0'`, and the types of the other properties must match as well.

**Pull request merged**